### PR TITLE
 Add rst:directive:option directive

### PR DESCRIPTION
**Pull request by tk0miya**
### Feature or Bugfix
- Feature

### Purpose
- Add `rst:directive:option` directive to describe the option for directives.
- This will help to mark up our document semantically

### CODE REVIEW

**Review by stephenfin — APPROVED** [review]

File: doc/usage/restructuredtext/domains.rst [hunk 1]
```diff
@@ -1419,6 +1419,31 @@ The reStructuredText domain (name **rst**) provides the following directives:
...
+   will be rendered as:
+
+       .. rst:directive:: toctree
+          :noindex:
+
+          .. rst:directive:option:: caption: caption of ToC
+
+          .. rst:directive:option:: glob
+
+   The ``type`` option can be given to describes the type of option value.
```
Comment (shimizukawa) [inline doc/usage/restructuredtext/domains.rst]: I want to have a example like:
```
.. rst:directive:: toctree

   .. rst:directive:option:: maxdepth
      :type: integer
```

**Review by shimizukawa — APPROVED** [review]
LGTM with nits

File: doc/usage/restructuredtext/domains.rst (hunk 1 above)
Comment (tk0miya) [reply doc/usage/restructuredtext/domains.rst]: done

**Comment by tk0miya** [comment]
Merged. Thank you for reviewing!

**Pull request merged**